From 28f2638230328bfcacc9ea947f8f5d7f65f067a4 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Wed, 14 Mar 2018 18:34:59 +0800
Subject: [PATCH 3511/5242] MLK-17804-2: dts: change gpio pin for qm mipi csi

commit  0b807d39ca90fd7af1c353fee3b50902ed9870b4 from
https://source.codeaurora.org/external/imx/linux-imx.git

Replace mipi csi gpio pin with generic gpio pin.

Reviewed-by: sandor.yu <sandor.yu@nxp.com>
Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit 36b7f6948052d4687fb11b86ec0403869133fc80)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts |   28 +++++++---------------
 1 file changed, 9 insertions(+), 19 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
index 7539b5b..8c3dd44 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
@@ -532,19 +532,15 @@
 			>;
 		};
 
-		pinctrl_mipi_csi0_gpio: mipicsi0gpiogrp{
+		pinctrl_gpio1: mipi_csi_en_rst{
 			fsl,pins = <
-				SC_P_MIPI_CSI0_GPIO0_00_MIPI_CSI0_GPIO0_IO00	0x00000021
-				SC_P_MIPI_CSI0_GPIO0_01_MIPI_CSI0_GPIO0_IO01	0x00000021
+				SC_P_MIPI_CSI0_GPIO0_00_LSIO_GPIO1_IO27		0x00000021
+				SC_P_MIPI_CSI0_GPIO0_01_LSIO_GPIO1_IO28		0x00000021
+				SC_P_MIPI_CSI1_GPIO0_00_LSIO_GPIO1_IO30		0x00000021
+				SC_P_MIPI_CSI1_GPIO0_01_LSIO_GPIO1_IO31		0x00000021
 			>;
 		};
 
-		pinctrl_mipi_csi1_gpio: mipicsi1gpiogrp{
-			fsl,pins = <
-				SC_P_MIPI_CSI1_GPIO0_00_MIPI_CSI1_GPIO0_IO00	0x00000021
-				SC_P_MIPI_CSI1_GPIO0_01_MIPI_CSI1_GPIO0_IO01	0x00000021
-			>;
-		};
 	};
 };
 
@@ -1043,15 +1039,9 @@
 	status = "okay";
 };
 
-&gpio0_mipi_csi0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_mipi_csi0_gpio>;
-	status = "okay";
-};
-
-&gpio0_mipi_csi1 {
+&gpio1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_mipi_csi1_gpio>;
+	pinctrl-0 = <&pinctrl_gpio1>;
 	status = "okay";
 };
 
@@ -1069,7 +1059,7 @@
 		clock-names = "capture_mclk";
 		mclk = <27000000>;
 		mclk_source = <0>;
-		pwn-gpios = <&gpio0_mipi_csi0 0 GPIO_ACTIVE_HIGH>;
+		pwn-gpios = <&gpio1 27 GPIO_ACTIVE_HIGH>;
 		virtual-channel;
 		port {
 			max9286_0_ep: endpoint {
@@ -1094,7 +1084,7 @@
 		clock-names = "capture_mclk";
 		mclk = <27000000>;
 		mclk_source = <0>;
-		pwn-gpios = <&gpio0_mipi_csi1 0 GPIO_ACTIVE_HIGH>;
+		pwn-gpios = <&gpio1 30 GPIO_ACTIVE_HIGH>;
 		virtual-channel;
 		port {
 			max9286_1_ep: endpoint {
-- 
1.7.9.5

